plugin/restore/mysql/mysql.go

```diff
@@ -83,6 +88,167 @@ func (r *Restore) RestorePITR(ctx context.Context, fullBackup *bufio.Scanner, bi
+}
+
+// backupSorter implements interface sort.Interface
+type backupSorter []backupComparator // nolint
+
+func (b backupSorter) Len() int      { return len(b) }           // nolint
+func (b backupSorter) Swap(i, j int) { b[i], b[j] = b[j], b[i] } // nolint
+
+// Newer binlog filename/position will be in front of older ones.
+// nolint
+func (b backupSorter) Less(i, j int) bool {
+	if b[i].binlogIndex > b[j].binlogIndex {
+		return true
+	}
+	return b[i].binlogInfo.Position > b[j].binlogInfo.Position
+}
+
+// parse the numeric extension part of the binlog file names, e.g., binlog.000001 -> 1
+func parseBinlogFileNameIndex(filename string) (int64, error) {
+	parts := strings.Split(filename, ".")
+	if len(parts) != 2 {
+		return -1, fmt.Errorf("the filename %q is not a valid binlog file name format", filename)
+	}
+	index, err := strconv.ParseInt(parts[1], 10, 64)
+	if err != nil {
+		return -1, fmt.Errorf("failed to parse string[%s] to integer, error[%w]", parts[1], err)
+	}
+	return index, nil
+}
+
+// Parse the binlog at position, and return the timestamp in the binlog event.
+// The current mechanism is by invoking mysqlbinlog and parse the output string.
+// Maybe we should parse the raw binlog header to get better documented structure?
+// nolint
+func (r *Restore) parseBinlogEventTimestamp(ctx context.Context, binlogInfo *api.BinlogInfo, binlogDir string) (int64, error) {
+	args := []string{
+		path.Join(binlogDir, binlogInfo.FileName),
+		fmt.Sprintf("--start-position %d", binlogInfo.Position),
+		// This will trick mysqlbinlog to output the binlog event header followed by a warning message telling that
+		// the --stop-position is in the middle of the binlog event.
+		// It's OK, since we are only parsing for the timestamp in the binlog event header.
+		fmt.Sprintf("--stop-position %d", binlogInfo.Position+1),
+	}
+	var buf bytes.Buffer
+	cmd := exec.CommandContext(ctx, r.mysqlbinlog.GetPath(), args...)
+	cmd.Stderr = os.Stderr
+	cmd.Stdout = &buf
+
+	if err := cmd.Run(); err != nil {
+		return -1, err
+	}
+
+	timestamp, err := parseBinlogEventTimestampImpl(buf.String(), binlogInfo.Position)
+	if err != nil {
+		return timestamp, fmt.Errorf("failed to parse binlog event timestamp, filename[%s], position[%d], error[%w]", binlogInfo.FileName, binlogInfo.Position, err)
+	}
+
+	return timestamp, nil
+}
+
+func parseBinlogEventTimestampImpl(output string, startPosition int64) (int64, error) {
+	lines := strings.Split(output, "\n")
+	// The mysqlbinlog output will contains a line "# at 12345", where 12345 is the --start-position param.
+	// The next line will be started as "#220425 17:32:13", which is the timestamp we are looking for.
+	atPos := fmt.Sprintf("# at %d", startPosition)
+	for i, line := range lines {
+		if strings.HasPrefix(line, atPos) {
+			timestampLine := lines[i+1]
+			year, err := strconv.Atoi("20" + timestampLine[1:3])
+			if err != nil {
+				return -1, err
+			}
+			month, err := strconv.Atoi(timestampLine[3:5])
```

Comment:
This approach has a lot of error handling. Can we use time.Parse()?

	date, error := time.Parse("060102 15:04:05", "220421 04:05:07")
	fmt.Println(date, error)